lib/utils/compose_ts.ts

```diff
@@ -588,9 +588,6 @@ async function getTokenForPreviousRepos(
 	const { authorizePush, getPreviousRepos } = await import('./compose');
 	const sdk = getBalenaSdk();
 	const previousRepos = await getPreviousRepos(sdk, docker, logger, appId);
-	if (!previousRepos || previousRepos.length === 0) {
-		return '';
```

Comment:
I had added this `if` test as an "optimization" but I misread the implementation of `authorizePush()`: it should be called even if `previousRepos` is an empty array.